Src/PerformanceCollector/Filtering/CollectionConfiguration.cs

```diff
@@ -0,0 +1,331 @@
+            CollectionConfigurationError[] documentStreamErrors;
+            this.CreateDocumentStreams(out documentStreamErrors, timeProvider, previousDocumentStreams ?? new DocumentStream[0]);
+
+            // check performance counters
+            CollectionConfigurationError[] performanceCounterErrors;
+            this.CreatePerformanceCounters(out performanceCounterErrors);
+
+            errors = metricErrors.Concat(documentStreamErrors).Concat(performanceCounterErrors).ToArray();
+
+            foreach (var error in errors)
+            {
+                error.Data["ETag"] = this.info.ETag;
+            }
+        }
+
+        public IEnumerable<OperationalizedMetric<RequestTelemetry>> RequestMetrics => this.requestTelemetryMetrics;
+
+        public IEnumerable<OperationalizedMetric<DependencyTelemetry>> DependencyMetrics => this.dependencyTelemetryMetrics;
+
+        public IEnumerable<OperationalizedMetric<ExceptionTelemetry>> ExceptionMetrics => this.exceptionTelemetryMetrics;
+
+        public IEnumerable<OperationalizedMetric<EventTelemetry>> EventMetrics => this.eventTelemetryMetrics;
+
+        public IEnumerable<OperationalizedMetric<TraceTelemetry>> TraceMetrics => this.traceTelemetryMetrics;
+
+        public IEnumerable<OperationalizedMetric<MetricValue>> MetricMetrics => this.metricMetrics;
+
+        /// <summary>
+        /// Telemetry types only (handled by QuickPulseTelemetryProcessor).
+        /// </summary>
+        public IEnumerable<Tuple<string, AggregationType>> TelemetryMetadata => this.telemetryMetadata;
+
+        /// <summary>
+        /// Metric type only (handled by QuickPulseMetricProcessor).
+        /// </summary>
+        public IEnumerable<Tuple<string, AggregationType>> MetricMetadata => this.metricMetadata;
+
+        /// <summary>
+        /// Document streams (handled by QuickPulseTelemetryProcessor).
+        /// </summary>
+        public IEnumerable<DocumentStream> DocumentStreams => this.documentStreams; 
+        
+        public string ETag => this.info.ETag;
+
+        /// <summary>
+        /// Gets a list of performance counters.
+        /// </summary>
+        /// <remarks>
+        /// Performance counter name is stored in OperationalizedMetricInfo.Projection.
+        /// </remarks>
+        public IEnumerable<Tuple<string, string>> PerformanceCounters => this.performanceCounters;
+
+        private static void AddMetric<TTelemetry>(
+          OperationalizedMetricInfo metricInfo,
+          List<OperationalizedMetric<TTelemetry>> metrics,
+          out CollectionConfigurationError[] errors)
+        {
+            errors = new CollectionConfigurationError[] { };
+
+            try
+            {
+                metrics.Add(new OperationalizedMetric<TTelemetry>(metricInfo, out errors));
+            }
+            catch (Exception e)
+            {
+                // error creating the metric
+                errors =
+                    errors.Concat(
```

Comment:
Yeah, it is ugly... could be List<>... This is not a performance-critical path, so probably no big deal,  but let me look at it. Thanks for pointing it out, this is a left-over from the early days when slopes were still snowy...